### PR TITLE
implement the interactions in impress slideshow.

### PR DESCRIPTION
**Pull request by hfiguiere**
* Resolves: # 
* Target version: master 

### Summary

This implement the interactions in impress slideshow.

Requires 
- [x] https://gerrit.libreoffice.org/c/core/+/172847
- [x] https://gerrit.libreoffice.org/c/core/+/173648

~~Includes that enable the UI (not necessary for this patch, useful for testing it) 
https://github.com/CollaboraOnline/online/pull/9836~~

### TODO

- [x] Seems that there are issue with "bookmark"
- [x] Non navigation interactions are no-op

### Checklist

- [x] I have run `make prettier-write` and formatted the code.
- [x] All commits have Change-Id
- [x] I have run tests with `make check`
- [x] I have issued `make run` and manually verified that everything looks okay
- [x] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

**Comment by hfiguiere** [comment]
LOKit patch merged

File: browser/src/slideshow/engine/SlideShowNavigator.ts [hunk 1]
```diff
@@ -243,8 +261,62 @@ class SlideShowNavigator {
...
+		} else if (aEvent.button === 2) {
+			this.switchSlide(-1, false);
+		}
+	}
+
+	_onExecuteInteraction(action: ClickAction) {
+		if (action) {
+			switch (action.action) {
+				case 'prevpage':
+					this.displaySlide(
```
Comment (mcecchetti) [inline browser/src/slideshow/engine/SlideShowNavigator.ts:296]: easier to call `this.switchSlide(-1, true)`;  `displaySlide` already takes care to check that the passed slide index is valid

File: browser/src/slideshow/engine/SlideShowNavigator.ts [hunk 2]
```diff
@@ -243,8 +261,62 @@ class SlideShowNavigator {
...
+						this.currentSlide ? this.currentSlide - 1 : 0,
+						true,
+					);
+					break;
+				case 'nextpage':
+					this.dispatchEffect();
```
Comment (mcecchetti) [inline browser/src/slideshow/engine/SlideShowNavigator.ts:302]: if the goal is to jump to next slide and not to start next effect, it should be invoked `this.switch(1, true)`

File: browser/src/slideshow/engine/SlideShowNavigator.ts [hunk 3]
```diff
@@ -129,6 +129,24 @@ class SlideShowNavigator {
...
 	}
 
+	goToSlideAtBookmark(bookmark: string) {
+		NAVDBG.print(
+			'SlideShowNavigator.goToSlideAtBookmark: ' +
+				bookmark +
+				' current index: ' +
+				this.currentSlide,
+		);
+		for (let i = 0; i < this.theMetaPres.numberOfSlides; i++) {
```
Comment (mcecchetti) [inline browser/src/slideshow/engine/SlideShowNavigator.ts:139]: I wonder if it could make more sense to iterate on core and set the `bookmark` parameter to the slide hash value instead of the slide name, so we save the need to retrieve and forward the slide name.
iterate on core or iterate on online ?  @eszkadev, @quikee ?

File: browser/src/slideshow/engine/SlideShowNavigator.ts (hunk 1 above)
Comment (hfiguiere) [reply browser/src/slideshow/engine/SlideShowNavigator.ts]: done

File: browser/src/slideshow/engine/SlideShowNavigator.ts (hunk 2 above)
Comment (hfiguiere) [reply browser/src/slideshow/engine/SlideShowNavigator.ts]: done

File: browser/src/slideshow/engine/SlideShowNavigator.ts [hunk 4]
```diff
@@ -243,8 +261,62 @@ class SlideShowNavigator {
...
+					this.goToFirstSlide();
+					break;
+				case 'lastpage':
+					this.goToLastSlide();
+					break;
+				case 'bookmark':
+					this.goToSlideAtBookmark(action.bookmark);
+					break;
+				case 'stoppresentation':
+					this.presenter.exitSlideshowWithWarning();
```
[the remark this answers is not in the document]
Comment (hfiguiere) [reply browser/src/slideshow/engine/SlideShowNavigator.ts:311]: I used the same thing as `endPresentation`. Maybe I should just call `endPresentation(true)` ?

[the remark this answers is not in the document]
Comment (mcecchetti) [reply browser/src/slideshow/engine/SlideShowNavigator.ts:311]: Better to call `SlideShowNavigator.quit` so `currentIndex` is cleared

[the remark this answers is not in the document]
Comment (hfiguiere) [reply browser/src/slideshow/engine/SlideShowNavigator.ts]: done

**Comment by eszkadev** [comment]
@hfiguiere please rebase and solve conflict

**Comment by hfiguiere** [comment]
rebased

**Comment by eszkadev** [comment]
@hfiguiere it stil shows conflict here

**Comment by hfiguiere** [comment]
uh I rebased the wrong branch... let me try again.

**Comment by hfiguiere** [comment]
rebased

File: browser/src/slideshow/engine/SlideShowNavigator.ts [hunk 5]
```diff
@@ -244,8 +262,59 @@ class SlideShowNavigator {
 	}
 
 	private clickHandler(aEvent: MouseEvent) {
-		if (aEvent.button === 0) this.dispatchEffect();
-		else if (aEvent.button === 2) this.switchSlide(-1, false);
+		if (aEvent.button === 0) {
```
Comment (eszkadev) [inline browser/src/slideshow/engine/SlideShowNavigator.ts:265]: apart of click we should also handle "hover" so we can use "hand" mouse pointer to indicate interactive element for the user

**Comment by eszkadev** [comment]
I tried with current core but I didn't manage to trigger interactive action in the test document (jump to last page). Am I missing something?

**Comment by eszkadev** [comment]
in test sample:
- on 1st slide I click "go to last slide" -> NOK - I go to the next one
- on 2nd slide I click "go to 5 slide" -> OK - I go to 5th
